python/lsst/ctrl/mpexec/cmdLineFwk.py

```diff
@@ -66,6 +76,145 @@
 _LOG = logging.getLogger(__name__.partition(".")[2])
 
 
+@dataclass
+class _OutputChainedCollectionInfo:
+
+    def __init__(self, registry: Registry, name: str):
+        self.name = name
+        try:
+            self.chain = list(registry.getCollectionChain(name))
+            self.exists = True
+        except MissingCollectionError:
+            self.chain = []
+            self.exists = False
+
+    name: str
+    exists: bool
+    chain: List[Tuple[str, DatasetTypeRestriction]]
+
+
+@dataclass
+class _OutputRunCollectionInfo:
+
+    def __init__(self, registry: Registry, name: str):
+        self.name = name
+        try:
+            actualType = registry.getCollectionType(name)
+            if actualType is not CollectionType.RUN:
+                raise TypeError(f"Collection '{name}' exists but has type {actualType.name}, not RUN.")
+            self.exists = True
+        except MissingCollectionError:
+            self.exists = False
+
+    name: str
+    exists: bool
+
+
+@dataclass
```

Comment:
Why is this a dataclass? It seems like a normal class with a bunch of code in it.

Reply:
It doesn't need to be; I just tend to start prototyping things as dataclasses because I like to think about data structures before methods.  These classes were the last thing I was working on before I had to step away from it, and they just never got cleaned up (or documented, as I'm sure you noticed).